### PR TITLE
test: Retry when 'Network access is disabled' in phantomjs

### PR DESCRIPTION
**Pull request by stefwalter**
phantomjs and QtWebkit seems to unpredictably return a failure
with 'Network access is disabled'. In the cases where we receive
this, we retry the last test

### CODE REVIEW

**Comment by dperpeet** [comment]
In the corresponding Qt bug description I believe often the Qt state had to be somehow changed to recover from this. I'm not sure if a simple reload works, as the test failures indicate:
```
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/base1/patternfly.css
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/shell/index.css
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/system/system.css
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/base1/jquery.js
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/base1/cockpit.js
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/*/po.js
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/system/system.js
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/domain/domain.js
Resource Error: Network access is disabled. http://10.111.117.115:9090/cockpit/@10.111.112.103/performance/performance.js
Wrote TestShutdownRestart-testBasic-FAIL.png
Journal extracted to TestShutdownRestart-testBasic-10.111.112.103-FAIL.log
Journal extracted to TestShutdownRestart-testBasic-10.111.117.115-FAIL.log
not ok 121 testBasic (check_shutdown_restart.TestShutdownRestart) duration: 182s
Traceback (most recent call last):
  File "./verify/check-shutdown-restart", line 106, in testBasic
    b2.enter_page("/system", host=m.address, reconnect=False)
  File "/build/cockpit/test/common/testlib.py", line 317, in enter_page
    self.wait_present("iframe.container-frame[name='%s'][data-loaded]" % frame)
  File "/build/cockpit/test/common/testlib.py", line 215, in wait_present
    return self.wait_js_func('ph_is_present', selector)
  File "/build/cockpit/test/common/testlib.py", line 209, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/common/testlib.py", line 722, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/common/testlib.py", line 745, in _invoke
    raise Error(res['error'])
Error: timeout
```

**Comment by stefwalter** [comment]
> In the corresponding Qt bug description I believe often the Qt state had to be somehow changed to recover from this. I'm not sure if a simple reload works, as the test failures indicate:

Yup. That was proven by my first try on this. Now it restarts entire tests.

**Comment by mvollmer** [comment]
check-networking-basic seems to fail often with a "health check failure" on centos-7 and rhel-7, always in the same place.  I'll look at that more closely.

File: test/common/testlib.py [hunk 1]
```diff
@@ -865,13 +892,15 @@ def not_ok(self, test, err):
...
         string = self._exc_info_to_string(err, test)
-        if self.naughty:
-            issue = self.naughty.check_issue(string)
+        if self.policy:
+            issue = self.policy.check_issue(string)
             if issue:
                 self.addSkip(test, "Known issue #{0}".format(issue))
                 return True
+            if self.policy.check_retry(string):
+                raise RetryError("Retrying due to failure of test hardness or framework")
```
Comment (mvollmer) [inline test/common/testlib.py]: s/hardness/harness/

**Comment by mvollmer** [comment]
So I tried to figure out whether the refactored networking tests have actually made things worse: #5964.  I think they have not, so we are probably looking at some real issue in the test infrastructure.

**Comment by stefwalter** [comment]
> I think they have not, so we are probably looking at some real issue in the test infrastructure.

The networking tests are flakey ... but they are not the source of 'Network access is disabled' style errors that have come with recent versions of phantomjs.

**Comment by mvollmer** [comment]
> The networking tests are flakey ...

Yes, I think I have found one source of the flakeyness: rolling back overly broad checkpoints.

> but they are not the source of 'Network access is disabled' style errors that have come with recent versions of phantomjs.

I agree.  Retrying is the only thing I can think of.  Sucks, but heh.

File: test/common/testlib.py (hunk 1 above)
Comment (stefwalter) [inline test/common/testlib.py]: If no other issue, this is worth fixing during merge?

File: test/common/testvm.py [hunk 2]
```diff
@@ -1232,6 +1232,10 @@ def kill(self):
                     self._domain.destroyFlags(libvirt.VIR_DOMAIN_DESTROY_DEFAULT)
             except:
                 pass
+            # print >> sys.stderr, "waiting for domain exit"
+            # if not self.event_handler.wait_for_undefined(self._domain, timeout_sec=60):
+            #     self.message("WARNING: waiting for libvirt machine undefine timed out: " + self._domain.name())
+            # print >> sys.stderr, "domain gone"
```
Comment (stefwalter) [inline test/common/testvm.py]: Whoops, this should be removed.

File: test/common/testlib.py (hunk 1 above)
Comment (dperpeet) [inline test/common/testlib.py]: I will fix during merge

File: test/common/testvm.py (hunk 2 above)
Comment (dperpeet) [inline test/common/testvm.py]: I will remove during merge

**Pull request closed without merging**